### PR TITLE
build,ci: use openwrtorg/sdk docker container

### PR DESCRIPTION
**Pull request by aparcar**
Instead of setting up the SDK for every single build, change build dir
from ~/build_dir to ~/openwrt.

As it now uses openwrtorg/sdk docker containers, there is no need to
keep the Dockerfile/README in .circleci.

The SDK docker images are based on https://github.com/openwrt/docker.

Signed-off-by: Paul Spooren <mail@aparcar.org>

### CODE REVIEW

**Comment by aparcar** [comment]
If desired I could change it to `nproc + 1`

**Comment by BKPepe** [comment]
In your commit, you are doing more changes than you are saying. May I ask you why did you remove README and Dockerfile?

**Comment by champtar** [comment]
@aparcar CircleCI only gives you 2 vCPUs, even if you see more than that with many commands (I remember we did some tests in the past)
maybe it changed but please test to see if it speed up anything

**Comment by aparcar** [comment]
> In your commit, you are doing more changes than you are saying. May I ask you why did you remove README and Dockerfile?

Updated in commit message

**Comment by aparcar** [comment]
> @aparcar CircleCI only gives you 2 vCPUs, even if you see more than that with many commands (I remember we did some tests in the past) maybe it changed but please test to see if it speed up anything

Right, I remember we tried this a long time ago with travis. I'm using the circleci cli which builds locally with 8 cores :) Changed it back to 3

**Comment by champtar** [comment]
@aparcar how often is the container updated ?

**Comment by aparcar** [comment]
> @aparcar how often is the container updated ?

Daily, based on snapshots...

**Comment by neheb** [comment]
No idea what this does exactly.

With the current setup, PKG_BUILD_DEPENDS can be tested since it starts all packages anew.

No idea if this changes that. In any case, @thess should be consulted as he did a lot of the CircleCI work.

**Comment by aparcar** [comment]
> With the current setup, PKG_BUILD_DEPENDS can be tested since it starts all packages anew.

What is PKG_BUILD_DEPENDS for?

**Comment by neheb** [comment]
It's for packages that are required for compilation but are not runtime dependencies. They're usually host packages that are not necessarily provided by the actual host.

Missing build dependencies tend to be found when compiling with multiple threads.

File: .circleci/config.yml [hunk 1]
```diff
@@ -125,14 +111,15 @@ jobs:
 
              for PKG in $PKGS ; do
                  echo_blue "===+ Building: $PKG"
-                 make "package/$PKG/compile" -j3 V=s
+                 # compiles as fast or verbose if failed on first try
+                 make "package/$PKG/compile" -j3 || make "package/$PKG/compile" V=s
```
Comment (champtar) [inline .circleci/config.yml]: It might hide // compilation issue, so not a fan
we can remove V=s if you want, but I'm curious how much speed we gain

**Comment by champtar** [comment]
@aparcar if you can rebase and answer my comment
thanks

File: .circleci/config.yml (hunk 1 above)
Comment (aparcar) [reply .circleci/config.yml]: Removed it.

**Comment by aparcar** [comment]
@champtar I don't see why these files conflict, I mean I removed them right?

**Comment by champtar** [comment]
Because we merged a PR modifying them, so git is nice enough to say that the files changed, and you should check again what you delete ;)

**Comment by aparcar** [comment]
Sorry missed to actually pull master :roll_eyes: The usign sign is great but requires to update the Docker images of sdk as well. I'll work on that later and hopefully have updates by tonight.

File: .circleci/config.yml [hunk 2]
```diff
@@ -2,11 +2,8 @@ version: 2.0
 jobs:
   build:
     docker:
-      - image: docker.io/openwrtorg/packages-cci:v1.0.5
+      - image: docker.io/openwrtorg/sdk:ath79-generic
```
Comment (champtar) [inline .circleci/config.yml]: Maybe we should have master somewhere in the version, like "sdk:master-ath79-generic" ?

**Comment by champtar** [comment]
Could you add a reference somewhere to https://github.com/openwrt/docker ?
Maybe just in the commit message so it's easier to find how this new image is built

File: .circleci/config.yml (hunk 2 above)
Comment (aparcar) [reply .circleci/config.yml]: Added, the current layout is actually ath79-generic-master

**Comment by neheb** [comment]
Is that httpd commit a mistake?

**Comment by aparcar** [comment]
> Is that httpd commit a mistake?

No it is to show that the PR works :) I'll remove it once approved.

**Comment by aparcar** [comment]
If there are no further concerns to test this, I'd like to merge this. I'll remove the apache2 commit just before that.

To keep the peace, please both ack this @jefferyto @diizzyy

**Comment by diizzyy** [comment]
The current version works although the downside is that it pulls down the SDK each time which consumes network traffic and increases overall run time.  As far as efficiency goes this looks like a step forward in the right direction however it also adds more complexity. I think it would be a wise idea to switch once OpenWrt have done the switch instead as it would also serve as a as an incentive to keep it working as it then would affect the project itself.

File: .circleci/config.yml [hunk 3]
```diff
@@ -60,45 +57,9 @@ jobs:
...
-      - run:
-          name: Prepare build_dir
-          working_directory: ~/build_dir
-          command: |
-             tar Jxf ~/sdk/$SDK_FILE --strip=1
-             touch .config
-             make prepare-tmpinfo scripts/config/conf
-             ./scripts/config/conf --defconfig=.config Config.in
-             make prereq
-             rm .config
```
Comment (jefferyto) [inline .circleci/config.yml]: I suggest incorporating the change from a6d5b5d1c1c6ea922afcfff1efac7fb55c9664c4 (or something like it) into
the `test-sdk` script in https://github.com/openwrt/docker/blob/master/.gitlab/ci/sdk.yml

**Comment by aparcar** [comment]
I chose to use GitHub CI instead and don't change anything about CircleCI in the future

**Pull request closed without merging**